core/state/statedb.go

```diff
@@ -87,6 +87,18 @@ func (self *StateDB) GetLogs(hash common.Hash) vm.Logs {
 	return self.logs[hash]
 }
 
+func (self *StateDB) GetAllLogs() *map[common.Hash]vm.Logs {
```

Comment:
Could you detail why these two methods are (`GetAllLogs` and `SetAllLogs`)  needed? I don't see them used in any part of the code.